lib/v2/chinawriter/index.js

```diff
@@ -0,0 +1,83 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const iconv = require('iconv-lite');
+const jschardet = require('jschardet');
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+
+const decodeBuffer = (buffer) => iconv.decode(buffer, jschardet.detect(buffer).encoding);
```

Comment:
I've tested all the entries in the docs. All of them can be decoded properly using the default `responseType`. `iconv-lite` isn't needed for this site.

Beside that, you can load the response buffer as `utf-8` and check the attribute of `$('meta[charset]')` or `$('meta[http-equiv="Content-Type"]')` if the response header does not provide charset.